bundle/jsky.app.ot/src/main/java/jsky/app/ot/tpe/GemsGuideStarWorker.java

```diff
@@ -116,24 +120,35 @@ public void setInterrupted() {
     }
 
     /**
+     * AGS failed to run, so clear the AGS results.
+     */
+    public void clearResults() {
+        apply(tpe.getContext(), 0.0, GuideGroup.AutomaticInitial());
+    }
+
+    /**
      * Apply the results of the guide star search to the current observation,
      * setting the selected position angle and guide probe.
      */
     public void applyResults(GemsGuideStars gemsGuideStars) {
-        applyResults(tpe.getContext(), gemsGuideStars);
+        apply(tpe.getContext(), gemsGuideStars.pa().toDegrees(), GuideGroup.createActive(gemsGuideStars.guideGroup().getAll()));
     }
 
-    public static void applyResults(final TpeContext ctx, final GemsGuideStars gemsGuideStars) {
-        SPInstObsComp inst = ctx.instrument().orNull();
+    private static void apply(final TpeContext ctx, final double paInDegrees, final GuideGroup autoGroup) {
+        final SPInstObsComp inst = ctx.instrument().orNull();
         if (inst != null) {
-            inst.setPosAngleDegrees(gemsGuideStars.pa().toDegrees());
+            inst.setPosAngleDegrees(paInDegrees);
             ctx.instrument().commit();
         }
 
-        TargetObsComp targetObsComp = ctx.targets().orNull();
+        final TargetObsComp targetObsComp = ctx.targets().orNull();
         if (targetObsComp != null) {
-            TargetEnvironment env = targetObsComp.getTargetEnvironment();
-            targetObsComp.setTargetEnvironment(env.setPrimaryGuideGroup(gemsGuideStars.guideGroup()));
+            final TargetEnvironment envOld = targetObsComp.getTargetEnvironment();
+            final GuideEnvironment gEnvOld = envOld.getGuideEnvironment();
+
+            final GuideEnvironment gEnvNew = gEnvOld.setGroup(0, autoGroup).setPrimaryIndex(0);
```

Comment:
Yes... it seems that the `GemsGuideStarSearchController.add` method is what is invoked to add GeMS asterisms, which calls a different `GemsGuideStarWorker.applyResults` method that does not use this `apply` under the hood.

It seems like there is a pretty weird and serious bug, though, in the way that the manual `GemsGuideStarWorker.applyResults` method puts the manual results in the `TargetEnvironment`: essentially, the table of GeMS results has a primary associated with it, I believe, but that primary is also used as the primary in the list of `GuideGroup` in the `TargetEnvironment`, which is replaced.

I do not know what the desired effects should be for a manual search. I would _think_ that we would want to just add a new manual `GuideGroup` with the results to the end of the list of `GuideGroups` and then mark it as the primary, but I am not sure. I have marked this as being broken in `GemsGuideStarWorker` for now with a TODO, but it might merit filing a separate JIRA issue since this seems incorrect.

There is also a bug now about determining column widths using the table utils that I know is my fault. It just issues a type warning, but it will have to be fixed. I've made a note to do so, but I think this needs to be done in another branch off develop.